srml/system/src/lib.rs

```diff
@@ -560,6 +588,7 @@ impl<T: Trait> Module<T> {
 	/// Remove temporary "environment" entries in storage.
 	pub fn finalize() -> T::Header {
 		ExtrinsicCount::kill();
+		Self::update_fee_multiplier();
```

Comment:
anyone reviewing this should check + see the big picture from this part of the code.